assert/assertions.go

```diff
@@ -202,6 +213,68 @@ func isNumericType(t reflect.Type) bool {
 	return t.Kind() >= reflect.Int && t.Kind() <= reflect.Complex128
 }
 
+// listsAreEqualValues gets whether two lists(arrays, slices) are equal, or if their
+// values are equal.
+//
+// This function should only be used by ObjectsAreEqualValues.
+func listsAreEqualValues(expected, actual interface{}) bool {
+	expectedValue := reflect.ValueOf(expected)
+	actualValue := reflect.ValueOf(actual)
```

Comment:
These reflect calls aren't needed until after the length checking, can we move them down.